tests/fft_test.py

```diff
@@ -47,6 +47,15 @@ def _get_fftn_test_axes(shape):
     axes.append((-index,))
   return axes
 
+def _get_fftn_test_s(shape, axes):
+  s_list = [None]
+  if axes is None or len(axes) == 0:
+    return s_list
+  s_list.append(tuple(shape[i] for i in axes))
+  s_list.append(tuple(shape[::-1][i]+1 for i in axes))
+  s_list.append(tuple(shape[i]+i for i in axes))
```

Comment:
This logic results in repeated entries, which leads to an error when running a wider suite of tests:
```
raise DuplicateTestNameError(test_class_name, new_name, original_name)
absl.testing.parameterized.DuplicateTestNameError: Duplicate parameterized test name in FftTest: generated test name 'testFftn_inverse=True_real=True_shape=int32[2,3,4]_axes=(-3,)_s=(5,)' (generated from 'testFftn_inverse=True_real=True_shape=int32[2,3,4]_axes=(-3,)_s=(5,)') already exists. Consider using named_parameters() to give your tests unique names and/or renaming the conflicting test method.
```
We'll need to fix this before we can merge.

Reply:
@JoostvDoorn You can have a look at https://github.com/google/jax/pull/6343/commits/9291632ea331a4b149ebd6e798505b4c50b64a67 where I tried adding the same as this PR.

The `testFftnWithShape` specifically tests fftn with explicit shapes.

Reply:
@peterroelants Thanks, I used your version of this function this is much simpler.